src/background/services/paymentSession.ts

```diff
@@ -264,8 +284,20 @@ export class PaymentSession {
       );
     };
 
+    if (!this.rate) {
+      // this.rate is set when adjustAmount begins. this.amount is set only after first successful adjustAmount
+      throw new Error('Unexpected: adjustAmount not yet ready');
+    }
     if (this.canContinuePayment) {
       this.timeout = setTimeout(async () => {
+        if (!this.amount) {
+          await this.adjustAmount(this.rate);
```

Comment:
This call here will reuse the existing probing promise instead of starting a new one.